src/daemon/daemon.cpp

```diff
@@ -2726,6 +3024,49 @@ void mp::Daemon::create_vm(const CreateRequest* request,
     prepare_future_watcher->setFuture(QtConcurrent::run(make_vm_description));
 }
 
+bool mp::Daemon::delete_vm(InstanceTable::iterator vm_it, bool purge, DeleteReply& response)
+{
+    auto& [name, instance] = *vm_it;
+    auto* erase_from = purge ? &deleted_instances : nullptr; // to begin with
+    auto instances_dirty = false;
+
+    if (!vm_instance_specs[name].deleted)
+    {
+        mpl::log(mpl::Level::debug, category, fmt::format("Deleting instance: {}", name));
+        erase_from = &operative_instances;
+        if (instance->current_state() == VirtualMachine::State::delayed_shutdown)
+            delayed_shutdown_instances.erase(name);
+
+        mounts[name].clear();
+        instance->shutdown();
```

Comment:
I'm pretty sure this function is a refactoring, but we don't really need to (try to) shutdown the instance if we are purging in the same operation.  It might make sense to move this into the `if (!purge)` block below.

Reply:
Yes, this was transported from [here](https://github.com/canonical/multipass/blob/ef40bb593eb7db4e76969be303f54733598d2aa5/src/daemon/daemon.cpp#L2127). Should we address that in this PR?

Reply:
Oh, right, another case of just moving existing code. (I really wish there was some way to tell that code is only moved in a diff, but I digress...).  I prefer not to change it, but I don't want us to forget about it either.  This would actually fit more in line with the `stop --force` stuff we need to do.  Let's leave it as is in this PR and I will figure out some way to not forget about it .:slightly_smiling_face: